src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4662,19 +4771,29 @@ private void CheckPossibleNullReceiver(BoundExpression receiverOpt)
 #endif
                 var resultType = _resultType.TypeSymbol;
                 if ((object)resultType != null &&
-                    _resultType.GetValueNullableAnnotation().IsAnyNullable() &&
-                    !resultType.IsValueType)
+                    _resultType.GetValueNullableAnnotation().IsAnyNullable())
                 {
-                    ReportDiagnostic(ErrorCode.WRN_NullReferenceReceiver, receiverOpt.Syntax);
-                    int slot = MakeSlot(receiverOpt);
-                    if (slot > 0)
+                    bool isValueType = resultType.IsValueType;
+                    if (isValueType && !allowValueType)
```

Comment:
>if (isValueType && !allowValueType) [](start = 20, length = 35)

It feels like this check should also explicitly filter out value types that are not ```Nullable<T>```.